test failures related to hdf5 lib

## Expected behavior ##

<!-- A clear and concise description of what you want to do and what you think should happen. (Code to reproduce the behavior can be added below). -->
The tests all pass at all times.


## Actual behavior ##

<!-- What happened instead. Add as much detail as you can. Include (copy and paste) stack traces and any output. -->

Scheduled CI failed, see https://github.com/Becksteinlab/zarrtraj/actions/runs/13220516432/job/36904931628 with errors pointing at the Python hdf5 package:

On macOS
```
E   ImportError: dlopen(/Users/runner/miniconda3/envs/zarrtraj-test/lib/python3.10/site-packages/h5py/defs.cpython-310-darwin.so, 0x0002): Symbol not found: _H5Pget_fapl_ros3
E     Referenced from: <0BB57518-61A3-3E6E-B925-56055D74DA28> /Users/runner/miniconda3/envs/zarrtraj-test/lib/python3.10/site-packages/h5py/defs.cpython-310-darwin.so
E     Expected in:     <AD1C1EA3-9334-3911-B862-F17F58553E3B> /Users/runner/miniconda3/envs/zarrtraj-test/lib/libhdf5.310.5.0.dylib
```

On Ubuntu
```
 ImportError while loading conftest '/home/runner/work/zarrtraj/zarrtraj/zarrtraj/tests/conftest.py'.
zarrtraj/__init__.py:7: in <module>
    from .ZARR import ZARRH5MDReader, ZARRMDWriter
zarrtraj/ZARR.py:139: in <module>
    from .utils import *
zarrtraj/utils.py:8: in <module>
    from kerchunk.hdf import SingleHdf5ToZarr
/usr/share/miniconda3/envs/zarrtraj-test/lib/python3.10/site-packages/kerchunk/hdf.py:17: in <module>
    import h5py
/usr/share/miniconda3/envs/zarrtraj-test/lib/python3.10/site-packages/h5py/__init__.py:33: in <module>
    from . import version
/usr/share/miniconda3/envs/zarrtraj-test/lib/python3.10/site-packages/h5py/version.py:15: in <module>
    from . import h5 as _h5
h5py/h5.pyx:1: in init h5py.h5
    ???
E   ImportError: /usr/share/miniconda3/envs/zarrtraj-test/lib/python3.10/site-packages/h5py/defs.cpython-310-x86_64-linux-gnu.so: undefined symbol: H5Pset_fapl_ros3
```


## Code to reproduce the behavior ##

See https://github.com/Becksteinlab/zarrtraj/actions/runs/13220516432

## What to do?

Not sure – possibly we need to pin hdf5 ??

--- a/devtools/conda-envs/test_env.yaml
+++ b/devtools/conda-envs/test_env.yaml
@@ -12,6 +12,7 @@ dependencies:
   - zarr>=2.11.0
   - kerchunk>=0.2.6
   - h5py>=3.11.0
+  - hdf5=1.12.1
 
   ### AWS dependencies ###
   # AWS reading/writing

